### PR TITLE
CLI: change layout from an option to a positional argument

### PR DESCRIPTION
**Pull request by kecnry**
### Description

This pull request changes the `--layout` option in the CLI to a positional (required) argument.  So `jdaviz FILENAME --layout=cubeviz` is now `jdaviz cubeviz FILENAME`.  This also removes `'default'` as an option for layout until it is properly supported.  In doing all of this, the CLI outputs change for the following:

**BEFORE**
`jdaviz`:

```
Usage: jdaviz [OPTIONS] FILENAME
Try 'jdaviz --help' for help.

Error: Missing argument 'FILENAME'.
```

`jdaviz --help`:

```
Usage: jdaviz [OPTIONS] FILENAME

  Start a Jdaviz application instance with data loaded from FILENAME.

Options:
  --layout [default|cubeviz|specviz|mosviz|imviz]
                                  Configuration to use on application startup
                                  [default: default]
  --browser TEXT                  Browser to use for application  [default:
                                  default]
  --theme [light|dark]            Theme to use for application  [default:
                                  light]
  --verbosity [debug|info|warning|error]
                                  Verbosity of the application  [default:
                                  info]
  --hotreload / --no-hotreload    Whether to enable hot-reloading of the UI
                                  (for development)
  --version                       Show the version and exit.
  --help                          Show this message and exit.
```

**AFTER**
`jdaviz`:
```
Usage: jdaviz [OPTIONS] {cubeviz|specviz|mosviz|imviz} FILENAME
Try 'jdaviz --help' for help.

Error: Missing argument '{cubeviz|specviz|mosviz|imviz}'. Choose from:
	cubeviz,
	specviz,
	mosviz,
	imviz
```

`jdaviz cubeviz`:

```
Usage: jdaviz [OPTIONS] {cubeviz|specviz|mosviz|imviz} FILENAME
Try 'jdaviz --help' for help.

Error: Missing argument 'FILENAME'.
```

`jdaviz --help`:

```
Usage: jdaviz [OPTIONS] {cubeviz|specviz|mosviz|imviz} FILENAME

  Start a Jdaviz application instance with data loaded from FILENAME.

Options:
  --browser TEXT                  Browser to use for application  [default:
                                  default]
  --theme [light|dark]            Theme to use for application  [default:
                                  light]
  --verbosity [debug|info|warning|error]
                                  Verbosity of the application  [default:
                                  info]
  --hotreload / --no-hotreload    Whether to enable hot-reloading of the UI
                                  (for development)
  --version                       Show the version and exit.
  --help                          Show this message and exit.
```

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [x] Are two approvals required? Branch protection rule does not check for the second approval. If a second approval is not necessary, please apply the `trivial` label.
- [x] Do the proposed changes actually accomplish desired goals? Also manually run the affected example notebooks, if necessary.
- [x] Do the proposed changes follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [x] Are tests added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [x] Are docs added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [x] Did the CI pass? If not, are the failures related?
- [x] Is a change log needed? If yes, is it added to `CHANGES.rst`?
- [x] Is a milestone set?
- [x] After merge, any internal documentations need updating (e.g., JIRA, Innerspace)?

### CODE REVIEW

**Comment by camipacifici** [comment]
Behavior looks good to me!

File: jdaviz/cli.py
Comment (pllim) [inline jdaviz/cli.py:20]: Is there no way to deprecate this first rather than straight out removing it? Maybe this has been discussed while I am out and I missed it.

Comment (kecnry) [reply jdaviz/cli.py:20]: I'm open to any suggestions.  Its definitely possible to still support `--layout` by renaming the positional argument to something else and checking to make sure if `--layout` is provided, its the same as the positional argument... but since the positional one is required, that seems like overkill.  If we switch the order so that the positional argument for layout is after the filename, then I think something like that becomes much more sensible, but it feels less natural to me to have `jdaviz /path/to/file cubeviz` than `jdaviz cubeviz path/to/file`.

Comment (pllim) [reply jdaviz/cli.py:20]: I was actually thinking one step more backward, as in holding off `layout` as positional, but issuing a message for `--layout`, which means everyone using the CLI will see this message for at least one release, before the API changes. Although given the small amount of users we have, maybe this is overkill?

Comment (kecnry) [reply jdaviz/cli.py:20]: That's definitely an option too... I'm just not sure how to make a message visible enough (for any use that is just launching from habit, a print to the terminal might easily go unnoticed).  I did just test issuing the old syntax in this PR though, and it gives the following (which isn't maybe perfectly ideal, but at least does kind of point the user in the right direction to noticing the change?):

```
Usage: jdaviz [OPTIONS] {cubeviz|specviz|mosviz|imviz} FILENAME
Try 'jdaviz --help' for help.

Error: No such option: --layout
```

File: CHANGES.rst
Comment (pllim) [inline CHANGES.rst:28]:
```suggestion
  (``jdaviz cubeviz path/to/file``). [#1252]
```

File: jdaviz/cli.py
Comment (pllim) [inline jdaviz/cli.py:21]: Are we deleting the "default" config? Seems like it is not possible to be used from CLI with this patch.

**Review by pllim — COMMENTED** [review]
I have no idea how MAST is calling their own layout so would be nice to hear from @havok2063 that this won't break MAST.

File: jdaviz/cli.py
Comment (pllim) [reply jdaviz/cli.py:20]: Maybe what you have is good enough.

File: jdaviz/cli.py
Comment (kecnry) [reply jdaviz/cli.py:21]: Yes, @camipacifici suggested removing it entirely until it gets proper support/documentation (if we ever get around to that) to prevent users from finding it in its half-baked state.

**Comment by kecnry** [comment]
afaik, MAST does not make any use of the cli-interface, so this should not affect any custom configuration embedded on their side 🤞

File: jdaviz/cli.py
Comment (pllim) [reply jdaviz/cli.py:21]: But we still let people do it from `Application(...)` call?

Comment (kecnry) [reply jdaviz/cli.py:21]: right, should we remove that as well?  I'm guessing that's not as easy to stumble-upon though?

Comment (pllim) [reply jdaviz/cli.py:21]: Perhaps not, but we are carrying around dead code, so what's the point?

File: CHANGES.rst
Comment (pllim) [inline CHANGES.rst:27]: Isn't it the first argument to the executable?
```suggestion
- CLI now takes the layout as a required first positional argument
```

Comment (pllim) [reply CHANGES.rst:27]: Technically, `jdaviz` is the command, so `layout` would be the first positional argument to the command, so your suggestion works for me. I will approve after change log is clarified. Thanks!

**Comment by havok2063** [comment]
@pllim @kecnry we call Jdaviz via the following
```
app = Application(configuration=config)
helper = [Helper](app)
```
where `[Helper]` is the name of a specific dynamically determined viztool, i.e. Specviz, Cubeviz, etc.  `config` is a constructed configuration dictionary using one of the viztool configs.  We fallback on the default configuration when no valid file format / helper tool can be identified.  

We don't use the CLI so I think it's safe to remove it from there, but I would hesitate to remove the default configuration YAML file or disallow `Application()` calls.

**Review by pllim — APPROVED** [review]
Thanks!

**Comment by rosteen** [comment]
@kecnry I think the merge conflict is with your update of the filename for the example cube.

**Review by rosteen — APPROVED** [review]
Looks good to me. I used the Github interface to resolve the merge conflict, which resulted in a merge commit - if you want to rebase locally to resolve it and force push instead feel free.

**Comment by pllim** [comment]
Luckily unlike `astropy`, we don't do backports, so let's push the "Squash and merge" button. 😉

**Pull request merged**
